src/js/popup.js

```diff
@@ -352,10 +352,13 @@ const Logic = {
   },
 
   getAssignmentObjectByContainer(userContextId) {
-    return browser.runtime.sendMessage({
-      method: "getAssignmentObjectByContainer",
-      message: { userContextId }
-    });
+    if (userContextId) {
```

Comment:
```suggestion
    if (!userContextId) {
      return {};
    }
```
Returning early reduces nesting and also in this case makes the 'diff noise' less too :)